packages/zowe-explorer/src/dataset/actions.ts

```diff
@@ -1057,16 +1065,19 @@ export async function submitJcl(datasetProvider: api.IZoweTree<api.IZoweDatasetT
  *
  * @param node The node/member that is being submitted
  * @param ownsJob Whether the current user profile owns this job
+ * @param fileName When submitting local JCL, use the document file name to submit
  * @returns Whether the job submission should continue.
  */
-async function confirmJobSubmission(node: api.IZoweTreeNode, ownsJob: boolean): Promise<boolean> {
+export async function confirmJobSubmission(node: api.IZoweTreeNode, ownsJob: boolean, fileName?: string): Promise<boolean> {
     ZoweLogger.trace("dataset.actions.confirmJobSubmission called.");
+
     const showConfirmationDialog = async (): Promise<boolean> => {
+        const jclName = node != null ? node.getLabel().toString() : fileName.toString().substring(fileName.lastIndexOf("\\") + 1);
```

Comment:
Works great @JasonIBM ! just had some feedback on the code :)

1. for the `node != null` part, this can be changed to `node !== null` 
2. for the call to `fileName.toString().substring(...)`, I see the `fileName` property is optional in the function, I would check for it to exist first before accessing its function to prevent breakage, perhaps can do some other behavior if the parameter is not passed or is null

Reply:
I agree - since `fileName` is optional, we should use the optional chaining operator or it will raise a null-like access violation during Coverity scans. 

I'm indifferent about the `null` check, but `!= null` will cover both `null` & `undefined` for the `node` parameter, so either way works 😋